Point set shape detection: Plane regularization

sgiraudot opened the pull request:
Small feature: https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Plane_regularization
* Not approved yet
* Partially tested

lrineau commented:
There are warnings on Windows:
https://cgal.geometryfactory.com/CGAL/testsuite/results-4.9-Ic-1.shtml#Point_set_shape_detection_3

lrineau commented:
There are compilation errors in plugins of Polyhedron demo:
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.9-Ic-1/Polyhedron_Demo/TestReport_afabri_x64_Cygwin-Windows8_MSVC2013-Release-64bits.gz
> test\Polyhedron_Demo\Plugins\Point_set\Point_set_shape_detection_plugin.cpp(11) : fatal error C1083: Cannot open include file: 'CGAL/Plane_regularization.h': No such file or directory

sgiraudot commented:
All should be fixed now.

(Sorry about the Polyhedron demo part, I always forget to propagate changes of API into the related plugin...)

sgiraudot commented:
This PR was pre-approved by @palliez (https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Plane_regularization). I took his latest remarks into account.

I also fixed some bugs and added an executable for the testsuite.

sgiraudot commented:
There were 2 warnings of implicit conversion of double to FT which I corrected. Everything else seems to work fine (the errors on the `IntelCompiler-17` platform do not seem related to this PR).

sloriot commented:
Wiki updated

lrineau commented on Installation/changes.html:
```diff
@@ -269,6 +269,13 @@ <h2 id="release4.8">Release 4.8 </h2>
+      algorithm: <code>CGAL::regularize_planes()</code>. This allows the user
+      to favor parallelism, orthogonality, coplanarity and/or axial
+      symmetry between detected planes.</li>
+  </ul>
```
The PR is merged in `master` (4.9) but the modification to the `changes.html` is in the section about CGAL-4.8!
@sloriot @sgiraudot

lrineau replied:
Done by PR #1062.
